include/OpenSpaceToolkit/Astrodynamics/Conjunction/Messages/CCSDS/CDM.hpp

```diff
@@ -0,0 +1,321 @@
+            State               state ;
+
+            // Covariance Matrix
+
+            // String              covarianceComment ;
+            MatrixXd            covarianceMatrix ;  // Usually defined in RTN Frame
+
+        } ;
+
+                                CDM                                         (   const   CDM::Header&                aHeader,
+                                                                                const   CDM::RelativeMetadata&      aRelativeMetadata,
+                                                                                const   Array<CDM::Metadata>&       aMetadataArray,
+                                                                                const   Array<CDM::Data>            aDataArray                                  ) ;
+
+        friend std::ostream&    operator <<                                 (           std::ostream&               anOutputStream,
+                                                                                const   CDM&                        anCDM                                       ) ;
+
+        bool                    isDefined                                   ( ) const ;
+
+        CDM::Header             getHeader                                   ( ) const ;
+
+        CDM::RelativeMetadata   getRelativeMetadata                         ( ) const ;
+
+        Array<CDM::Metadata>    getMetadataArray                            ( ) const ;
+
+        Array<CDM::Data>        getDataArray                                ( ) const ;
+
+        CDM::Metadata           getObjectMetadataAt                         (   const   Size&                       anIndex                                     ) const ;
+
+        CDM::Data               getObjectDataAt                             (   const   Size&                       anIndex                                     ) const ;
+
+        // Header getters
+
+        String                  getCCSDSCDMVersion                          ( ) const ;
+
+        Instant                 getCreationDate                             ( ) const ;
+
+        String                  getOriginator                               ( ) const ;
+
+        String                  getMessageFor                               ( ) const ;
+
+        String                  getMessageId                                ( ) const ;
+
+        // Relative Metadata getters
+
+        Instant                 getTCA                                      ( ) const ;
+
+        Length                  getMissDistance                             ( ) const ;
+
+        Position                getRelativePosition                         ( ) const ;
+
+        Velocity                getRelativeVelocity                         ( ) const ;
+
+        Real                    getCollisionProbability                     ( ) const ;
+
+        String                  getCollisionProbabilityMethod               ( ) const ;
```

Comment:
Just checking: no enum for this?

Reply:
Not right now indeed. From Annex E in https://public.ccsds.org/Pubs/508x0b1e2c2.pdf, it seems like there are a couple of methods supported. Yet, unfortunately a lot of users don't necessarily respect the format, so I think keeping it general with a `String` is fine, especially because this is informational for now (at least in the first MVP to fetch and display Conjunction)